raiden/raiden_service.py

```diff
@@ -311,13 +312,22 @@ def stop(self):
         wait_for = [self.alarm]
         wait_for.extend(self.protocol.greenlets)
         wait_for.extend(self.greenlet_task_dispatcher.stop())
-        gevent.wait(wait_for)
+        # We need a timeout to prevent an endless loop from trying to
+        # contact the disconnected client
+        gevent.wait(wait_for, timeout=self.shutdown_timeout)
 
         # Filters must be uninstalled after the alarm task has stopped. Since
         # the events are polled by a alarm task callback, if the filters are
         # uninstalled before the alarm task is fully stopped the callback
         # `poll_blockchain_events` will fail.
-        self.blockchain_events.uninstall_all_event_listeners()
+        #
+        # We need a timeout to prevent an endless loop from trying to
+        # contact the disconnected client
+        try:
+            with gevent.Timeout(self.shutdown_timeout):
+                self.blockchain_events.uninstall_all_event_listeners()
```

Comment:
If the node is offline, the filters are already gone, I don't think there is much use for waiting for it to come back online.

Reply:
The `uninstall_all_event_listeners` function calls `uninstall` on every `Filter` which results in a RPC-call which will block.

Reply:
Yes, the point is that the RPC-call is useless if the ethereum node is offline. Ideally the code would just give up on uninstalling if the ethereum node is offline, instead of waiting for the timeout.

Note: I'm assuming that raiden and the ethereum node are running on the same machine, this actually makes sense for communication through the network, since we can't know for sure if the node is offline or the network is hiccuping.

Reply:
I agree with the reasoning but I'm not sure how to handle that. Might be easiest to add a `connected` property to the `JSONRPCClient` and set it accordingly in the `_checkNodeConnection` decorator. Do you think that would work?

Regarding your assumption: I'd say it's a valid assumption to have ethereum node and raiden on the same machine. Did we discuss that?

Reply:
> Do you think that would work?

Yes, although it's a bit convoluted, can't you expose a version of the rpc client that does not retry by default?

> Did we discuss that?

I don't recall this being made an explicit assumption.

Reply:
The problem with exposing different kinds of client is that most of the interaction happens over the `BlockchainService` object. Another option would be to introduce a per-method `retry` parameter. However I think that complicates stuff more than using the timeout in the end.